model-optimizer/extensions/middle/StridedSliceNormalizer.py

```diff
@@ -0,0 +1,238 @@
+            # insert blank values for ellipsis unrolling and extending
+            for mask_name in StridedSlice.get_all_mask_names():
+                node[mask_name] = np.insert(node[mask_name], insertation_start_idx, [0] * num_insertations).astype(int)
+
+    @staticmethod
+    def unroll_ellipsis_for_inputs(graph: Graph, node: Node, ellipsis_start: int, num_ellipsis_ext: int):
+        node_name = node.soft_get('name', node.id)
+        for i, slice_name in enumerate(('begin', 'end', 'strides')):
+            i += 1
+            placeholder_arr = np.zeros(num_ellipsis_ext) if i != 3 else np.ones(num_ellipsis_ext)
+            placeholder_node = Const(graph, {'name': node_name + '/const_to_unroll_{}_ellipsis'.format(slice_name),
+                                             'value': int64_array(placeholder_arr)}).create_node()
+
+            concat_in_ports_count = 3 if ellipsis_start != 0 else 2
+            concat = Concat(graph, {'axis': 0, 'name': node_name + '/concat_{}'.format(slice_name),
+                                    'in_ports_count': concat_in_ports_count}).create_node()
+
+            if ellipsis_start != 0:
+                split = create_op_with_const_inputs(graph, VariadicSplit, {1: 0, 2: int64_array([ellipsis_start, -1])},
+                                                    {'name': node_name + '/split_for_{}_ellipsis'.format(slice_name),
+                                                     'out_ports_count': 2})
+                node.in_port(i).get_connection().set_destination(split.in_port(0))
+
+                concat.in_port(0).connect(split.out_port(0))
+                concat.in_port(1).connect(placeholder_node.out_port(0))
+                concat.in_port(2).connect(split.out_port(1))
+            else:
+                concat.in_port(0).connect(placeholder_node.out_port(0))
+                node.in_port(i).get_connection().set_destination(concat.in_port(1))
+
+            concat.out_port(0).get_connection().set_destination(node.in_port(i))
+
+    @staticmethod
+    def extend_inputs(node: Node, num_insertations: int):
+        graph = node.graph
+        node_name = node.soft_get('name', node.id)
+
+        for i, slice_name in enumerate(('begin', 'end', 'strides')):
+            i += 1
+            if node.in_port(3).disconnected():
+                break  # if strides are not specified no need for extending
+
+            # for strides blank ones should be inserted
+            blank_values_arr = np.zeros(num_insertations) if i != 3 else np.ones(num_insertations)
+            blank_values_node = Const(graph, {'name': node_name + '/extend_{}_const'.format(slice_name),
+                                             'value': int64_array(blank_values_arr)}).create_node()
+
+            if node.in_port(i).get_source().node.soft_get('type') == 'Concat':
+                # concat already exists
+                concat = node.in_port(i).get_source().node
+                last_in_port = concat['in_ports_count']
```

Comment:
The attribute "in_ports_count" is not updated when you add new input ports. The proper way is to get `max(node.in_ports().keys())`

Reply:
corrected